flipp/src/app/components/AppContainer.js

```diff
@@ -1,7 +1,31 @@
+"use client";
 import CardContainer from "./CardContainer";
 import Header from "./Header";
 import ButtonContainer from "./ButtonContainer";
+import { useState } from "react";
 export default function AppContainer() {
+  const [flip, setFlip] = useState(false);
+  const [currentCard, setCurrentCard] = useState(0);
+  function flipHandleClick() {
```

Comment:
These functions below are passed as props to the components down in the hierarchy to handle click events